### PR TITLE
Hostgroup code improvement + coverage(BZ1372917)

### PR DESCRIPTION
**Pull request by oshtaier**
```
nosetests tests/foreman/ui/test_hostgroup.py
......F....
----------------------------------------------------------------------
Ran 11 tests in 821.341s

FAILED (failures=1)
```
One failure related to environment issue

### CODE REVIEW

**Comment by oshtaier** [comment]
Do not squash commits please

File: tests/foreman/ui/test_hostgroup.py [hunk 1]
```diff
@@ -145,48 +209,77 @@ def test_positive_update(self):
...
+            )
+            self.hostgroup.search_and_click(name)
+            self.hostgroup.click(
+                tab_locators['hostgroup.tab_operating_system'])
+            self.assertEqual(
+                self.hostgroup.wait_until_element(
+                    locators['hostgroups.architecture_value']).text,
+                arch.name
+            )
+            self.hostgroup.click(locators['hostgroups.operating_system_clear'])
```
Comment (abalakh) [inline tests/foreman/ui/test_hostgroup.py]: Have you thought about some kind of update() logic like 'if parameter_value is None, then click 'clear' locator for parameter_name'? Or it was withdrawn as it won't be universal? To clear text fields we can pass empty string '', for such fields - None. Just a quick idea without deep analysis :)

Comment (oshtaier) [inline tests/foreman/ui/test_hostgroup.py]: here is an idea to click exactly 'cross' icon, not to clear a value in select list

Comment (abalakh) [inline tests/foreman/ui/test_hostgroup.py]: We had a small chat about this idea and agreed to try it out. Please do not merge this PR until further updates.

**Comment by oshtaier** [comment]
@abalakh addressed your previous comment

**Pull request merged**